src/averageAges.js

```diff
@@ -55,7 +68,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(child => onlyWithSon
+    ? people.some(
+      mom => child.mother === mom.name) && child.sex === 'm'
+    : people.some(mom => child.mother === mom.name)
+  );
+
```

Comment:
And here again, you duplicate:
`people.some(mom => child.mother === mom.name)` let's create a separate callback for this